packages/actions/test/unit/security.test.ts

```diff
@@ -89,70 +110,281 @@ describe("Security", () => {
+                const userFirebaseCredentials = GithubAuthProvider.credential(token)
+                await signInToFirebaseWithCredentials(userApp, userFirebaseCredentials)
+                const user = getCurrentFirebaseAuthUser(userApp)
+                userId = user.uid
 
-        it("should throw an error if an authenticated user tries to read another user's data", async () => {
-            // login as user2
-            await signInWithEmailAndPassword(userAuth, user2.data.email, user2Pwd)
-            assert.isRejected(getDocumentById(userFirestore, commonTerms.collections.users.name, user1.uid))
+                await signOut(userAuth)
+
+                // @todo how to revoke the token programmatically?
+                await signInToFirebaseWithCredentials(userApp, userFirebaseCredentials)
+            })
+            /// @note A malicious user should not be able to create multiple malicious accounts
+            /// to spam a ceremony
+            // @todo requires adding the checks to the cloud function
+            it("should prevent a user with a non reputable GitHub account from authenticating to the Firebase", async () => {})
+            /// @note If a coordinator disables an account, this should not be allowed to authenticate
+            it("should prevent a disabled account from loggin in (OAuth2)", async () => {
+                const auth = createOAuthDeviceAuth({
+                    clientType,
+                    clientId,
+                    scopes: ["gist"],
+                    onVerification: simulateOnVerification
+                })
+                const { token } = await auth({
+                    type: tokenType
+                })
+                // Get and exchange credentials.
+                const userFirebaseCredentials = GithubAuthProvider.credential(token)
+                await signInToFirebaseWithCredentials(userApp, userFirebaseCredentials)
+
+                const user = getCurrentFirebaseAuthUser(userApp)
+                userId = user.uid
+                // Disable user.
+                const disabledRecord = await adminAuth.updateUser(user.uid, { disabled: true })
+                expect(disabledRecord.disabled).to.be.true
+
+                await signOut(userAuth)
+
+                await expect(signInToFirebaseWithCredentials(userApp, userFirebaseCredentials)).to.be.rejectedWith(
+                    "Firebase: Error (auth/user-disabled)."
+                )
+            })
+        }
+        /// @note It should not be possible to enumerate valid email addresses
+        /// using the error messages returned by the server (wrong email/ wrong password)
+        /// @todo This feature needs to be enabled in one account,
+        /// document this feature to prevent enumeration attacks
+        it.skip("should not allow a user to enumerate valid emails", async () => {
+            // @link https://cloud.google.com/identity-platform/docs/admin/email-enumeration-protection
```

Comment:
Open an issue to track doing this and adding to docs if outside of dev ops process

Reply:
https://github.com/privacy-scaling-explorations/groth-p0tion/issues/26